### PR TITLE
Handle sign in errors

### PR DESCRIPTION
**Pull request by aguzovatii**

### CODE REVIEW

**Review by Guzovatii-L — APPROVED** [review]

File: app/auth/signin/page.tsx [hunk 1]
```diff
@@ -3,46 +3,83 @@
 import { useState } from "react";
 import { signIn } from "next-auth/react";
 import Link from "next/link";
+import { useSearchParams } from "next/navigation";
```
Comment (DianaJacota) [inline app/auth/signin/page.tsx:6]:
```suggestion
```

File: app/auth/signup/page.tsx [hunk 2]
```diff
@@ -2,44 +2,82 @@
 
 import { useState } from "react";
 import { signIn } from "next-auth/react";
+import Link from "next/link";
+import { useSearchParams } from "next/navigation";
```
Comment (DianaJacota) [inline app/auth/signup/page.tsx:6]:
```suggestion
```

File: app/auth/signin/page.tsx [hunk 3]
```diff
@@ -69,14 +106,31 @@ export default function Signin() {
...
+        setErrorType(error);
+        return;
+      }
+
+      const ok = response.ok;
+      const redirectUrl = response.url;
+      if (ok) {
+        setSignedIn(ok);
+        setRedirectUrl(redirectUrl);
+      }
```
Comment (DianaJacota) [inline app/auth/signin/page.tsx:133]:
```suggestion
      if (response.ok) {
        setSignedIn(response.ok);
        setRedirectUrl(response.url);
      }
```

File: app/auth/signup/page.tsx [hunk 4]
```diff
@@ -66,14 +104,31 @@ export default function Signup() {
...
+        setErrorType(error);
+        return;
+      }
+
+      const ok = response.ok;
+      const redirectUrl = response.url;
+      if (ok) {
+        setSignedIn(ok);
+        setRedirectUrl(redirectUrl);
+      }
```
Comment (DianaJacota) [inline app/auth/signup/page.tsx:131]:
```suggestion
      if (response.ok) {
        setSignedIn(response.ok);
        setRedirectUrl(response.url);
      }
```

File: app/auth/signup/page.tsx [hunk 5]
```diff
@@ -66,14 +104,31 @@ export default function Signup() {
...
       username: username,
       password: password,
       callbackUrl: "/",
+    }).then((response) => {
+      const error = response.error;
+
+      if (error) {
+        setErrorType(error);
+        return;
+      }
```
Comment (DianaJacota) [inline app/auth/signup/page.tsx:124]:
```suggestion

      if (response.error) {
        setErrorType(response.error);
        return;
      }
```

File: app/auth/signin/page.tsx [hunk 6]
```diff
@@ -69,14 +106,31 @@ export default function Signin() {
...
       username: username,
       password: password,
       callbackUrl: "/",
+    }).then((response) => {
+      const error = response.error;
+
+      if (error) {
+        setErrorType(error);
+        return;
+      }
```
Comment (DianaJacota) [inline app/auth/signin/page.tsx:126]:
```suggestion

      if (response.error) {
        setErrorType(response.error);
        return;
      }
```

**Review by DianaJacota — APPROVED** [review]
Looks nice. Just a few optional comments from my side

**Pull request merged**